torch/distributed/_sharded_tensor/ops/embedding.py

```diff
@@ -141,11 +200,25 @@ def _handle_col_wise_sharding(input, world_size, weight, local_shard, pg):
         world_size: number of ranks.
         weight: shareded weight tensor.
         local_shard: col-wise shared local weight used for lookup.
+        max_norm: If given, each embedding vector with norm larger
+            than max_norm is renormalized to have norm max_norm.
+            Note: this will modify weight in-place.
+        norm_type: The p in the p-norm to compute for the max_norm option.
+        padding_idx: If specified, the entries at padding_idx do
+            not contribute to the gradient; therefore, the embedding
+            vector at padding_idx is not updated during training,
+            i.e. it remains as a fixed “pad”.
         pg: process group.
 
     Returns: final result of lookup.
     """
-    return _handle_col_wise_sharding_base(
+    if max_norm is not None:
+        # max_norm changes the weight in-place
+        local_shard = _handle_max_norm_col_wise(
+            max_norm, norm_type, local_shard, input, pg
+        )
```

Comment:
It seems like we are applying max_norm to the entire shard which doesn't seem right. The max_norm logic should be only applied to the embedding indices that we are looking up and not the whole shard.

Reply:
Yes... I thought about this. If we want to copy the exact behavior, we could end up adding more communication overhead because we need to first let each rank knows which IDs are being looked up. If you feel this is necessary, I can modify it.

Reply:
There is additional communication overhead but reduced compute overhead since you update only a few indices. But in general isn't this a correctness issue? In the unit tests can you compare the weights after the max_norm update? I think vanilla nn.embeddingbag only renormalizes the indices you look up and not the whole embedding bag.

Reply:
Yes, you are right. I did two things here:
1. I did the gather of input in this stage so that, no extra communication cost and we now only renorm the indices we look up. So this is correct now. I should have mentioned clearly in the update comment.
2. The unit test does compare the weight now. The only thing is that I need to first apply inputs from all ranks to the embedding/embedding bag so that the local will be renorm as sharded. Otherwise, the test would fai.